build-logic/documentation/src/main/groovy/gradlebuild/docs/GradleJavadocsPlugin.java

```diff
@@ -179,47 +180,62 @@ public JavadocSupport(Javadoc javadoc) {
         }
 
         public void setDestinationDir(File destinationDir) {
-            setProperty("setDestinationDir", destinationDir);
+            setProperty(javadoc, "setDestinationDir", destinationDir);
         }
+    }
 
-        private <T> void setProperty(String setterName, T value) {
-            if (GradleVersion.current().compareTo(GradleVersion.version("9.0")) < 0) {
-                setPropertyPreGradle9(setterName, value);
-            } else {
-                setPropertyPostGradle9(setterName, value);
-            }
+    /**
+     * TODO: Remove this workaround after Gradle 9
+     */
+    private static class CheckstyleSupport {
+        private final Checkstyle checkstyle;
+
+        public CheckstyleSupport(Checkstyle checkstyle) {
+            this.checkstyle = checkstyle;
         }
 
-        private void setPropertyPostGradle9(String setterName, Object value) {
-            try {
-                // Javadoc is Javadoc_Decorated, so it has set<Property>(Object) setter
-                javadoc.getClass().getMethod(setterName, Object.class).invoke(javadoc, value);
-            } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
-                throw new RuntimeException(e);
-            }
+        public void setClasspath(FileCollection classpath) {
+            setProperty(checkstyle, "setClasspath", classpath);
         }
+    }
+
+    private static  <T> void setProperty(Task task, String setterName, T value) {
+        if (GradleVersion.current().compareTo(GradleVersion.version("9.0")) < 0) {
+            setPropertyPreGradle9(task, setterName, value);
+        } else {
+            setPropertyPostGradle9(task, setterName, value);
+        }
+    }
+
+    private static void setPropertyPostGradle9(Task task, String setterName, Object value) {
+        try {
+            // Task is Task_Decorated, so it has set<Property>(Object) setter
+            task.getClass().getMethod(setterName, Object.class).invoke(task, value);
+        } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
+            throw new RuntimeException(e);
+        }
+    }
 
-        private void setPropertyPreGradle9(String setterName, Object value) {
-            try {
-                for (Method method : javadoc.getClass().getMethods()) {
-                    if (method.getParameters().length == 1
-                        && method.getName().equals(setterName)
-                        && method.getParameters()[0].getType().isAssignableFrom(value.getClass())) {
-                        method.invoke(javadoc, value);
-                        return;
-                    }
+    private static void setPropertyPreGradle9(Task task, String setterName, Object value) {
+        try {
+            for (Method method : task.getClass().getMethods()) {
+                if (method.getParameters().length == 1
+                    && method.getName().equals(setterName)
```

Comment:
Unfortunately not, since `value.getClass()` could be `ConfigurableFileCollection_Decorated` but we a method has a parameter `FileCollection`. That is why we do `isAssignable()`